### PR TITLE
Support for `given` 

### LINKED ISSUE #65: Document givens

### PR DESCRIPTION
**Pull request by pikinier20**
closes #65

### CODE REVIEW

File: src/main/scala/tests/givenSignatures.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package givenSignatures
+
+
+
+class GivenClass {
+    trait B
+    trait C[T]
+    val r: Int = 5
+    type R = Int
+    given R = r
```
Comment (abgruszecki) [inline src/main/scala/tests/givenSignatures.scala:12]: This given in the docs links to itself, but instead should link to `type R`.

File: src/main/scala/tests/givenSignatures.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        def compare(xs: List[T], ys: List[T]): Int = (xs, ys) match
+            case (Nil, Nil) => 0
+            case (Nil, _) => -1
+            case (_, Nil) => +1
+            case (x :: xs1, y :: ys1) =>
+                val fst = ord.compare(x, y)
+                if (fst != 0) fst else compare(xs1, ys1)
+    }
+
+    given IntOps as Int.type = Int
```
Comment (abgruszecki) [inline src/main/scala/tests/givenSignatures.scala:38]: This given should be displayed as `given IntOps as Int.type`, but instead is display as `given IntOps`.

**Review by abgruszecki — COMMENTED** [review]
Overall LGTM. One additional nitpick: looking at the code I'm not sure that anonymous "method" `given`-s display correctly. Example:
   ```scala
   given [T](using ord: Ord[T]) as Ord[List[T]]
   ```

File: src/main/scala/tests/givenSignatures.scala (hunk 2 above)
Comment (pikinier20) [reply src/main/scala/tests/givenSignatures.scala:38]: [info] method SymbolOps
[info] WARN: Unsupported type: TypeRef: Reflection.this.Symbol

Missing type is not bug of this PR, it's our incomplete type parsing

File: src/main/scala/tests/givenSignatures.scala (hunk 1 above)
Comment (pikinier20) [reply src/main/scala/tests/givenSignatures.scala:12]: Fixed

**Pull request merged**